js/index.js

```diff
@@ -16,4 +16,31 @@ for (var i = 0; i < skills.length; i++) {
     const skill =document.createElement('li');
     skill.innerText = skills[i];
     skillsList.appendChild(skill); 
-} 
\ No newline at end of file
+} 
+const messageForm = document.forms.leave_message;
+messageForm.addEventListener('submit', function (event) {
+    event.preventDefault();
+    const userName = event.target.usersName.value;
+    const userEmail = event.target.usersEmail.value;
+    const userMessage = event.target.usersMessage.value;
+    console.log('User Name:', userName);
+    console.log('User Email:', userEmail);
+    console.log('User Message:', userMessage);
+    const messageSection = document.getElementById('messages');
+    const messageList = messageSection.querySelector('ul');
+    messageForm.reset();
+    const newMessage = document.createElement('li');
+    newMessage.innerHTML = `
+        <a href="mailto:${userEmail}">${userName}</a>
+        <span>${userMessage}</span>
+    `;
+    const removeButton = document.createElement('button');
+    removeButton.innerText = 'remove';
+    removeButton.type = 'button';
+    removeButton.addEventListener('click', function () {
+        const entry = removeButton.parentNode;
+        entry.remove();
+    });
+    newMessage.appendChild(removeButton);
+    messageList.appendChild(newMessage);
+});
```

Comment:
In line 20 your trying to make a reference to a form that doesn't exist (Because you named your form as "Leave a message" in your HTML 
So if you try to log in console messageForm you're getting undefined.
Making a correct reference to the form, solves the issue.